docs/developer/architecture/coding-principles.md

```diff
@@ -26,27 +26,67 @@
 ## III. Simplicity
 
 1. Avoid layers of indirection when they are not needed (e.g. "pass-through methods").
-
-### IV. Testing
+2. Avoid needlessly wrapping objects, especially primitive datatypes.
+
+### IV. General Coding Style
+
+1. Use `var` instead of explicit types (helps with clarity)
+2. Avoid `final` in method args and local variables
+3. Use `final` in field declarations
+4. Avoid `static` fields except in constants or when absolutely necessary. (you should be able to provide a reason).
+5. Use interfaces to define shared constants
+6. Use "minimally required types" (or "smallest possible API"), e.g. use `ObjectMapper` instead of `TypeManager`
+   , or use a `String` instead of a more complex object containing the String, etc.
+7. Use either `public` members, which are documented and tested, or `private` members.
+8. Avoid package-private members, especially if only needed for testing
+9. Avoid `protected` members unless they're intended to be overridden.
+10. Use package-private classes if they're not needed outside the package, e.g. implementation classes
+11. Avoid using `enum`s for anything other than named integer enumerations.
+12. Avoid using static classes as much as possible. Exceptions to this are helper functions and test utils, etc. as well as static inner classes.
+13. Use only camel case and no prefixes for naming.
+14. Avoid unnecessary `this.` (exception: in builders)
+15. Use static imports, as long as it is still clear what's being done. For example,
```

Comment:
I commented this a lot of time, I'm absolutely in favor of static imports when the method name is self-explainatory (es. `Assertions.asserThat` -> `assertThat`), but not for other cases like `Stream.of` or `Optional.of`, they wouldn't be explainatory, moreover, if two static methods with the same name (as the `of` methods I just listed) are used in the same class they can't be both imported statically. So I would leave this decision driven by readability and code comprehension.

Reply:
>  if two static methods with the same name

that should be implicit though - it just won't compile anyway :)

> So I would leave this decision driven by readability and code comprehension

that is what I meant by "as long as it still clear...", but I can use your phrasing to be more precise.